wordpress-mobile/WordPress-iOS
# Makes some improvements to DomainSuggestionsTableViewController.

Fixes #17206


Make some improvements to `DomainSuggestionsTableViewController`.  This was created after [some suggestions](https://github.com/wordpress-mobile/WordPress-iOS/pull/17266#discussion_r723380409) by @frosty in a previous review.

This PR also sets the privacy protection on by default, for domains that support it.

## To test:

1. Go to the Domain Purchase flow.
2. Use the Domain Search regularly and make sure the suggestions come up as expected.
3. Go to My Site -> Domains and add a new domain
    a. Search for a domain that does not support privacy
    b. Proceed with the registration and make sure it succeeds
    c. Repeat the step listed in 3, and this time select a domain that support privacy
    d. Proceed with the registration and make sure it succeeds
    e. OPTIONALLY: in both cases, if you have access to the DARC, go there and make sure that the privacy protection is set accordingly
     

For the second test I recommend to insert these lines in `DomainsService.swift:53`, so that the suggestions service will always fail:

```swift
failure(NSError(domain: "testDomain", code: 1, userInfo: [:]))
return
```

1. Go to the Domain Purchase flow.
2. Make sure the no-results view is shown.

## Regression Notes

1. Potential unintended areas of impact

None as the domains flow is new.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

## PR submission checklist:
